src/scanner.hpp

```diff
@@ -26,27 +26,14 @@ class Scanner
   void add_token(TokenType type);
   void add_token(TokenType type, LiteralType literal);
   bool match(char key);
+
+  void report(int line, std::string_view message);
```

Comment:
Why does this need to be mutable? `void report(int line, std::string_view message) const;` otherwise.